example/traffic_mgmt/odp_traffic_mgmt.c

```diff
@@ -534,6 +556,44 @@ static int create_and_config_tm(void)
 	egress.egress_fcn  = tester_egress_fcn;
 
 	odp_tm_test = odp_tm_create("TM test", &requirements, &egress);
+	if (odp_tm_test == ODP_TM_INVALID) {
+		printf("Error: failed to create TM\n");
+		return -1;
+	}
+
+	if (odp_tm_capability(odp_tm_test, &tm_capa) != 0) {
+		printf("Error: failed to get tm capability");
+		return -1;
+	}
+
+	tm_shaper_min_rate = tm_capa.per_level[0].min_rate;
+	tm_shaper_max_rate = tm_capa.per_level[0].max_rate;
+	tm_shaper_min_burst = tm_capa.per_level[0].min_burst;
+	tm_shaper_max_burst = tm_capa.per_level[0].max_burst;
+
+	for (level = 1; level < tm_capa.max_levels; level++) {
+		odp_tm_level_capabilities_t *per_level =
+			&tm_capa.per_level[level];
+
+		if (per_level->min_rate > tm_shaper_min_rate)
+			tm_shaper_min_rate = per_level->min_rate;
+
+		if (per_level->min_burst > tm_shaper_min_burst)
+			tm_shaper_min_burst = per_level->min_burst;
+
+		if (per_level->max_rate < tm_shaper_max_rate)
+			tm_shaper_max_rate = per_level->max_rate;
+
+		if (per_level->max_burst < tm_shaper_max_burst)
+			tm_shaper_max_burst = per_level->max_burst;
+	}
+
+	if (tm_shaper_min_rate > tm_shaper_max_rate ||
+	    tm_shaper_min_burst > tm_shaper_max_burst) {
+		printf("Error: shaper min and max rates are non overlapping");
```

Comment:
I think this message is a bit unclear (I would expect max and min rates to not "overlap" (be the same)). Maybe something like this would be better: "Error: No shaper rate supported by all TM levels"